src/java/org/apache/cassandra/cql3/QueryOptions.java

```diff
@@ -53,6 +54,10 @@
 
     public static final CBCodec<QueryOptions> codec = new Codec();
 
+    private static final int UNSET_NOWINSEC = Integer.MIN_VALUE;
+    // Guava uint counterpart to the above for serialization purposes. This imposes a year 2186 max value
+    private static final int UNSET_NOWINSEC_GUAVA_UINT = UnsignedInteger.MAX_VALUE.intValue() - 1;
```

Comment:
Can't we use `UnsignedInteger.MAX_VALUE.intValue()`, without subtracting one?

Reply:
You want to protect against `UnsignedInteger.MAX_VALUE` becoming/being a sentinel value, so -1 should do the trick.

Reply:
Could you elaborate?

Reply:
MAX and MIN are often used in the code as sentinel values. You don't want sbdy passing you `UnsignedInteger.MAX_VALUE` to signal some special condition and accidentally mapping that to `UNSET_NOWINSEC`. So the -1 is to avoid that collision.

`MAX_VALUE-1` imo is a 'better' sentinel value, if such a thing can be said about any random value, because it's far less likely to collide and allows consumers to use the 'classic' MAX sentinel values they are used to.

But no other hard reasons as nowInSec won't be around MAX values until year 2106.

Reply:
This is not a good reason. This simply makes it look like we _do_ have some other sentinel value to future readers of the code, who will go hunting for one and fail to find it. Equally there can be more than one sentinel value - indeed, we have just introduced another one here! 

The only valid approach to sentinels is to be sure you know what they all are, assign them logically, and document them clearly. Padding your sentinels and hoping not to collide is just asking for problems, even more so if everyone followed this logic.

Reply:
Well I'll disagree somehow here. Putting some thought into trying to make your design friendlier is a perfectly good reason.Now if your ideas don't get traction then so be it. I don't feel strongly about it. I'll run it through CI next time I commit.

Reply:
I don't see what's the utility on `UNSET_NOWINSEC_GUAVA_UINT` here.

That constant is only used on `encode` and `decode` methods when serializing a `long` `nowInSec` as and unsigned integer. However, that serialization/derialization only happens when we have `Flag.NOW_IN_SECONDS`. That flag is only present if the `long` `nowInSec` is distinct to `UNSET_NOWINSEC`. 

So IIUC `UNSET_NOWINSEC_GUAVA_UINT` is never used, and it can be removed. Am I missing something?

Reply:
Correct. Good catch and confirmed with CI